app/src/main/java/br/com/dcarv/arcmovies/presentation/presenter/IPresenter.kt

```diff
@@ -0,0 +1,13 @@
+package br.com.dcarv.arcmovies.presentation.presenter
+
+import br.com.dcarv.arcmovies.presentation.view.IView
+
+/**
+ * Base interface for all Presenters
+ *
+ * @author Danilo Carvalho
+ */
+interface IPresenter {
+    fun attachView(view: IView)
+    fun detachView()
+}
```

Comment:
Why is it necessary to attach and detach the view from the presenter? What could happen if the view isn't detached when it's destroyed?

Reply:
It would leak. On a device configuration change, when the activities and fragment are destroyed and recreated, the presenter would keep a reference to the view if it was not detached. Also if we didn't attach the new view, the presenter would post updates to the old view, not the new one.